routers/api/v1/repo/issue.go

```diff
@@ -55,12 +60,30 @@ func ListIssues(ctx *context.APIContext) {
 		isClosed = util.OptionalBoolFalse
 	}
 
-	issues, err := models.Issues(&models.IssuesOptions{
-		RepoIDs:  []int64{ctx.Repo.Repository.ID},
-		Page:     ctx.QueryInt("page"),
-		PageSize: setting.UI.IssuePagingNum,
-		IsClosed: isClosed,
-	})
+	var issues []*models.Issue
+
+	keyword := strings.Trim(ctx.Query("q"), " ")
+	if strings.IndexByte(keyword, 0) > 0 {
```

Comment:
Change to `>=` or `!= -1` (Doesn't guard against `\x00` being the first byte)

Reply:
ok thanks, I didn't know about that.